app/controllers/analytics/districts_controller.rb

```diff
@@ -60,17 +60,19 @@ def set_organization_district
   end
 
   def set_cohort_analytics(period, prev_periods)
-    @cohort_analytics = set_analytics_cache(
-      analytics_cache_key_cohort(period),
-      @organization_district.cohort_analytics(period, prev_periods))
+    @cohort_analytics =
+      set_analytics_cache(analytics_cache_key_cohort(period)) do
+        @organization_district.cohort_analytics(period, prev_periods)
+      end
```

Comment:
Oh... oh dear... I see what happened here 😅

Reply:
In Haskell, everything is lazy by default.